bellman/src/groth16/tests/dummy_engine.rs

```diff
@@ -396,10 +396,12 @@ impl CurveProjective for Fr {
     type Affine = Fr;
     type Base = Fr;
 
-    fn batch_normalization(_: &mut [Self]) {}
+    fn batch_normalize(p: &[Self], q: &mut [Self::Affine]) {
```

Comment:
It might be worth implementing this as the default method on the trait. :thinking:

Reply:
Agreed; I'll add it to the next PR.